deploy.tf

```diff
@@ -381,10 +374,166 @@ EOF
+  name = "${var.prefix}k8s-etcd-fw"
+
+  droplet_ids = ["${digitalocean_droplet.k8s_etcd.id}"]
+
+  inbound_rule = [
+    {
+      protocol           = "tcp"
+      port_range         = "22"
+      source_addresses   = ["${digitalocean_droplet.k8s_master.ipv4_address_private}"]
+    },
+    {
+      protocol           = "tcp"
+      port_range         = "2379-2380"
+      source_addresses   = ["${digitalocean_droplet.k8s_master.ipv4_address_private}", "${digitalocean_droplet.k8s_worker.*.ipv4_address_private}"]
+    },
+  ]
+
+  outbound_rule = [
+    {
+      protocol                = "tcp"
+      port_range              = "all"
+      destination_addresses   = ["0.0.0.0/0", "::/0"]
+    },
+    {
+      protocol                = "udp"
+      port_range              = "all"
+      destination_addresses   = ["0.0.0.0/0", "::/0"]
+    },
+  ]
+
+}
+
+###############################################################################
 #
+# worker host inbound firewall rules
+#
+# TCP port 22 from master node for debugging over ssh
+# TCP port 10250 from Master nodes for Kubelet API exec and logs.
+# TCP port 10255 from Heapster Worker node for read-only Kubelet API.
+# TCP port ALL from Master & Worker Nodes for Intra-cluster communication (unnecessary if vxlan is used for networking)
+# UDP port 8285 from Master & Worker Nodes for flannel overlay network - udp backend. This is the default network configuration
+# UDP port 8472 from Master & Worker Nodes 	flannel overlay network - vxlan backend (only required if using flannel)
+###############################################################################
+
+resource "digitalocean_firewall" "k8s_worker" {
+  name = "${var.prefix}k8s-worker-fw"
+
+  droplet_ids = ["${digitalocean_droplet.k8s_worker.*.id}"]
+
+  inbound_rule = [
+    {
+      protocol           = "tcp"
+      port_range         = "all"
+      source_addresses   = ["${digitalocean_droplet.k8s_master.ipv4_address_private}", "${digitalocean_droplet.k8s_worker.*.ipv4_address_private}"]
+    },
+    {
+      protocol           = "tcp"
+      port_range         = "10250"
+      source_addresses   = ["${digitalocean_droplet.k8s_master.ipv4_address_private}", "${digitalocean_droplet.k8s_worker.*.ipv4_address_private}"]
+    },
+    {
+      protocol           = "tcp"
+      port_range         = "10255"
+      source_addresses   = ["${digitalocean_droplet.k8s_master.ipv4_address_private}"]
+    },
+    {
+      protocol           = "udp"
+      port_range         = "8285"
+      source_addresses   = ["${digitalocean_droplet.k8s_master.ipv4_address_private}", "${digitalocean_droplet.k8s_worker.*.ipv4_address_private}"]
+    },
+    {
+      protocol           = "udp"
+      port_range         = "8472"
+      source_addresses   = ["${digitalocean_droplet.k8s_master.ipv4_address_private}", "${digitalocean_droplet.k8s_worker.*.ipv4_address_private}"]
+    },
+    {
+      protocol           = "tcp"
+      port_range         = "80"
```

Comment:
Sorry for the late reply! Can we add `443` here?

Reply:
added